File: mongodb-manual/crud/11-query-find-01.py
#!/usr/bin/env python3
import pymongo
import pprint

client = pymongo.MongoClient("mongodb://localhost")
db = client.MyMongoManualPy

# SELECT * FROM inventory
for d in db.inventory.find(): pprint.pprint(d)

# SELECT * FROM inventory WHERE status = "D"
for d in db.inventory.find({"status": "D"}):
  pprint.pprint(d)

# SELECT * FROM inventory WHERE status in ("A", "D")
for d in db.inventory.find({"status": {"$in": ["A", "D"]}}):
  pprint.pprint(d)

# SELECT * FROM inventory WHERE status = "A" AND qty < 30
for d in db.inventory.find({"status": "A", "qty": {"$lt": 30}}):
  pprint.pprint(d)

# SELECT * FROM inventory WHERE status = "A" OR qty < 30
for d in db.inventory.find(
    {"$or": [{"status": "A"}, {"qty": {"$lt": 30}}]}):
  pprint.pprint(d)

# SELECT * FROM inventory WHERE status = "A"
#                           AND ( qty < 30 OR item LIKE "p%")
for d in db.inventory.find({
    "status": "A",
    "$or": [{"qty": {"$lt": 30}}, {"item": {"$regex": "^p"}}]}):
  pprint.pprint(d)

File: mongodb-manual/crud/11-query-find-01.js
db = db.getSiblingDB('MyMongoManual');

// SELECT * FROM inventory
db.inventory.find();

// SELECT * FROM inventory WHERE status = "D"
db.inventory.find( { status: "D" } )

// SELECT * FROM inventory WHERE status in ("A", "D")
db.inventory.find( { status: { $in: [ "A", "D" ] } } )

// SELECT * FROM inventory WHERE status = "A" AND qty < 30
db.inventory.find( { status: "A", qty: { $lt: 30 } } )

// SELECT * FROM inventory WHERE status = "A" OR qty < 30
db.inventory.find( { $or: [ { status: "A" }, { qty: { $lt: 30 } } ] } )

// SELECT * FROM inventory WHERE status = "A"
//                           AND ( qty < 30 OR item LIKE "p%")
db.inventory.find( {
  status: "A",
  $or: [ { qty: { $lt: 30 } }, { item: /^p/ } ]
} )
